custom_components/petsafe/__init__.py

```diff
@@ -147,6 +150,46 @@ async def handle_modify_schedule(call: ServiceCall) -> None:
         DOMAIN, SERVICE_MODIFY_SCHEDULE, handle_modify_schedule
     )
 
+    async def handle_feed(call: ServiceCall) -> None:
+        device_ids = call.data.get(ATTR_DEVICE_ID)
+        area_ids = call.data.get(ATTR_AREA_ID)
+        entity_ids = call.data.get(ATTR_ENTITY_ID)
+        amount = call.data.get(ATTR_AMOUNT)
+        slow_feed = call.data.get(ATTR_SLOW_FEED)
+        matched_devices = get_feeders_for_service(
+            hass, area_ids, device_ids, entity_ids
+        )
+
+        for device_id in matched_devices:
+            device = next(
+                d for d in await coordinator.get_feeders() if d.api_name == device_id
+            )
+            if device is not None:
+                await device.feed(amount, slow_feed, False)
+                await coordinator.async_request_refresh()
+
+    hass.services.async_register(DOMAIN, SERVICE_FEED, handle_feed)
+
+    async def handle_prime(call: ServiceCall) -> None:
+        device_ids = call.data.get(ATTR_DEVICE_ID)
+        area_ids = call.data.get(ATTR_AREA_ID)
+        entity_ids = call.data.get(ATTR_ENTITY_ID)
+        matched_devices = get_feeders_for_service(
+            hass, area_ids, device_ids, entity_ids
+        )
+
+        for device_id in matched_devices:
+            device = next(
+                d for d in await coordinator.get_feeders() if d.api_name == device_id
+            )
+            if device is not None:
+                # NB: DeviceSmartFeed.prime() synchronously updates state after priming.
```

Comment:
Well that's a bug in the petsafe library I missed, but no objections to moving this along in the meantime.